src/WebAuthn.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "./utils/Base64URL.sol";
+import "./P256.sol";
+
+/**
+ * Helper library for external contracts to verify WebAuthn signatures.
+ **/
+library WebAuthn {
+    function contains(
+        string memory substr,
+        string memory str,
+        uint256 location
+    ) internal pure returns (bool) {
+        bytes memory substr_bytes = bytes(substr);
+        bytes memory str_bytes = bytes(str);
+
+        uint256 substr_len = substr_bytes.length;
+        uint256 str_len = str_bytes.length;
+
+        for (uint256 i = 0; i < substr_len; i++) {
+            if (location + i >= str_len) {
+                return false;
+            }
+
+            if (substr_bytes[i] != str_bytes[location + i]) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    function checkAuthFlags(
+        bytes1 flags,
+        bool requireUserVerification
+    ) internal pure returns (bool) {
+        // Check the UP bit
+        if (flags & 0x01 != 0x01) {
+            return false;
+        }
+
+        // Check the UV bit
+        if (requireUserVerification && (flags & 0x04) != 0x04) {
```

Comment:
I think in Daimo usage we should pass this `requireUserVerification = false`. It should be the passkey authenticator's job to enforce such conditions (and the good ones we use seem to do it properly already), just leaving this here in case other users want it for any reason, as it also means we have feature parity for users of FCL's Webauthn implementation.